src/python/pants/core/target_types.py

```diff
@@ -858,9 +863,30 @@ async def package_archive_target(field_set: ArchiveFieldSet) -> BuiltPackage:
 # -----------------------------------------------------------------------------------------------
 
 
-class LockfileSourceField(SingleSourceField):
+class _IgnoreUnmatchedBuildFileGlobs(Enum):
+    ignore = "ignore"
+
+    @staticmethod
+    def as_unmatched_build_file_globs() -> UnmatchedBuildFileGlobs:
```

Comment:
There is no "ignore" in UnmatchedBuildFileGlobs AFAICT? So this seems pretty sketchy. We're lying to the typesystem, and basically to the reader...

We already had two enums to express glob match error behavior, now we have three. It just seems like overkill. How about we get rid of UnmatchedBuildFileGlobs and just use GlobMatchError everywhere, then we get the functionality we want *and* we get simpler code! I'd be perfectly happy with that tradeoff.

Reply:
It allows users to turn off any warn or error too, and that seems fine!

Reply:
OK, I can make that switch. That's the reason I didn't add the third option to the enum exposed as option as it would be a user visible change.. ;)
 
(we're only lying very locally here, but I agree it's at least semi-sketchy although API-wise it's fully compatible)

Reply:
Only because Python doesn't just let you shoot yourself in the foot but hands you a loaded gun... :)

I think simplifying here while also getting more functionality and fixing a bug is a win-win-win!